File: api/functional/flow-creator/src/main/java/com/emanuelvictor/api/functional/flowcreator/domain/repositories/AlternativeRepository.java
package com.emanuelvictor.api.functional.flowcreator.domain.repositories;

import com.emanuelvictor.api.functional.flowcreator.domain.entities.Alternative;
import com.emanuelvictor.api.functional.flowcreator.infrastructure.persistence.CrudRepository;

import java.util.List;

/**
 * @author Emanuel Victor
 * @version 1.0.0
 * @since 2.0.0, 01/01/2020
 */
public interface AlternativeRepository extends CrudRepository<Alternative, Long> {

    /**
     * @param previousId {@link Long}
     * @return {@link List}
     */
    List<Alternative> findByPreviousId(final Long previousId);
}

File: api/functional/flow-creator/src/main/kotlin/com/emanuelvictor/api/functional/flowcreator/domain/repositories/AlternativeRepository.kt
package com.emanuelvictor.api.functional.flowcreator.domain.repositories

import com.emanuelvictor.api.functional.flowcreator.domain.entities.alternative.Alternative
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

/**
 * @author Emanuel Victor
 * @version 1.0.0
 * @since 2.0.0, 01/01/2020
 */
interface AlternativeRepository : JpaRepository<Alternative, Long> {

    /**
     *
     */
    fun findByPath(path: String): Optional<Alternative>

}